---
title: "hodgkin_lymphoma_analysis"
output: html_notebook
---

Analyzing and plotting NanoString results for Hodgkin lymphoma project
Comparison of gene and protein values

```{r}

### load libraries, nCounter output raw data file, samples meta data file, counts/percentages for markers in total cells (DAPI) and tumor cells

library(openxlsx)
library(readxl)
library(DESeq2)
library(ggplot2)
library(dplyr)
library(pheatmap)
library(usefun)
library(ggplot2)
library(Kendall)
library(ggpubr)
library(tidyverse)
library(dendsort)
library(ggrepel)
library(mosaic)
library(gplots)
library(cowplot)
library(stringr)

rawcounts_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/nanostring/sequencing_results/RawData.csv"
normcounts_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/nanostring/sequencing_results/NormalizedData.csv"
samples_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/meta_data/HodgkinLymphoma_Samples__V1.xlsx"
clinical_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/meta_data/HodgkinLymphoma_Clinical__V1.xlsx"
#pDAPI_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/mpIF/tables/220720_all_marker_counts_per_sample.xlsx"
#pTumor_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/hodgkin_project/mpIF/tables/220720_tumor_functional_marker_counts_per_sample.xlsx"

rawcounts <- read.csv(rawcounts_path, header = TRUE, row.names = NULL, stringsAsFactors = FALSE, sep = ",")
normcounts <- read.csv(normcounts_path, header = TRUE, row.names = NULL, stringsAsFactors = FALSE, sep = ",")
samples <- read.xlsx(samples_path, colNames = TRUE, rowNames = FALSE)
clinical <- read.xlsx(clinical_path, colNames = TRUE, rowNames = FALSE)
#pDAPI <- read.xlsx(pDAPI_path, colNames = TRUE, rowNames = TRUE)
#pTumor_sample <- read_excel(pTumor_path, sheet = "marker_counts_by_CellDive_ID")
#pTumor_FOV <- read_excel(pTumor_path, sheet = "marker_counts_by_FOV_ID")
print("library and files loaded")
```

```{r}

### reformat NanoString count file to gene (x) by sample (y) matrix using Patient_ID, drop unwanted samples, create log and zscore version of this file

drop_cols <- c("X20190924_09631.G.cart1_PanelStandard_12.RCC", 
               "X20190924_09631.G.cart2_PanelStandard_12.RCC", 
               "X20190925_09631.G.cart3_PanelStandard_12.RCC", 
               "X20190925_09631.G.cart4_PanelStandard_12.RCC")

reformat <- function(ncounter_output_file, drop_cols, samples_meta, clinical_meta){
  ncounter_output_file <- ncounter_output_file %>% 
    filter(!row_number() %in% c(1, 2)) %>% 
    filter(Class.Name == "Endogenous") %>% 
    column_to_rownames(var = "Probe.Name") %>% 
    select(-c(1:18)) %>% 
    select(-all_of(drop_cols))
  new_col_names <- list()
  for (i in colnames(ncounter_output_file)){
    new_name <- str_split(i, "_")[[1]][3]
    new_name <- gsub('[.]', "_", new_name)
    new_col_names <- append(new_col_names, new_name)
  }
  colnames(ncounter_output_file) <- new_col_names
  # subset samples and clinical meta to include only samples with nanostring, remove NLPHL sample (HL_13) (n=40)
  samples_nanostring <- samples %>% 
    filter(!is.na(Nanostring_ID)) %>% 
    filter(!Patient_ID == "HL_13")
  clinical_nanostring <- clinical %>% 
    filter(Patient_ID %in% samples_nanostring$Patient_ID)
  rownames(clinical_nanostring) <- clinical_nanostring$Patient_ID
  # list of samples with nanostring data that will not be included in analysis
  drop <- outersect(samples_nanostring$Nanostring_ID, colnames(ncounter_output_file))
  # drop those samples from counts, reorder columns to match row order in meta_nanostring
  new_order <- samples_nanostring$Nanostring_ID
  ncounter_output_file <- ncounter_output_file %>% 
    select(-drop) %>% 
    select(new_order)
  # replace column names from nanostring ID to patient ID
  colnames(ncounter_output_file) <- samples_nanostring$Patient_ID
  # convert values in counts to numeric
  ncounter_output_file <- mutate_all(ncounter_output_file, function(x) as.numeric(as.character(x)))
  return(list(ncounter_output_file, clinical_nanostring))
}

reformat_rawcounts_temp <- reformat(ncounter_output_file = rawcounts, drop_cols = drop_cols, samples_meta = samples, clinical_meta = clinical)
reformat_rawcounts <- reformat_rawcounts_temp[[1]]
clinical_nanostring <- reformat_rawcounts_temp[[2]]
reformat_rawcounts_temp <- reformat(ncounter_output_file = normcounts, drop_cols = drop_cols, samples_meta = samples, clinical_meta = clinical)
reformat_normcounts <- reformat_rawcounts_temp[[1]]

print("gene by sample matrix created")
```

```{r}

### differential expression

#list of comparisons
clinical_nanostring$Diagnosis_abbrev2 <- clinical_nanostring$Diagnosis_abbrev
clinical_nanostring <- clinical_nanostring %>% 
  mutate(Diagnosis_abbrev2 = str_replace(Diagnosis_abbrev2, "MC", "Other")) %>% 
  mutate(Diagnosis_abbrev2 = str_replace(Diagnosis_abbrev2, "LR", "Other")) %>% 
  mutate(Diagnosis_abbrev2 = str_replace(Diagnosis_abbrev2, "Unclassified cHL", "Other")) %>% 
  relocate(Diagnosis_abbrev2, .after = Diagnosis_abbrev)
comp <- c("Sex", "Age_Diagnosis_Group", "Diagnosis_abbrev_2", "Clinical_Stage_Group", "Sample_type_abbrev", "EBV_final", "Treatment", "Bcell_Tcell", "PDL1_IHC", "PDL2_IHC", 
          "B2M_IHC", "MHCI_IHC", "MHCII_IHC", "CT2A_IHC")
#deseq2
identical(colnames(normcounts), rownames(clinical_nanostring))


dds <- DESeqDataSetFromMatrix(countData=normcounts, colData=clinical_nanostring, design=~EBV_final)
dds$Lesion_response <- relevel(dds$Lesion_response, "UT")
dds <- DESeq(dds)
res_nonCR <- results(dds, contrast = c("Lesion_response", "non_CR", "UT"))
res_CR <- results(dds, contrast = c("Lesion_response", "CR", "UT"))
res_nonCR_Ordered <- res_nonCR[order(res_nonCR$pvalue),]
res_CR_Ordered <- res_CR[order(res_CR$pvalue),]
res_nonCR_Ordered <- data.frame(GeneID = rownames(res_nonCR_Ordered), res_nonCR_Ordered)
res_CR_Ordered <- data.frame(GeneID = rownames(res_CR_Ordered), res_CR_Ordered)

#for now
dds <- DESeqDataSetFromMatrix(countData=counts, colData=meta_nanostring, design=~pop)
dds$pop <- relevel(dds$pop, "B-cell")
dds <- DESeq(dds)
res <- results(dds, contrast = c("pop", "T-cell", "B-cell"))
res_Ordered <- res[order(res$padj),]
res_Ordered <- data.frame(GeneID = rownames(res_Ordered), res_Ordered)



#counts <- as.matrix(counts)
normcounts[, 1:ncol(normcounts)] <- sapply(normcounts[, 1:ncol(normcounts)], as.numeric)

comp_relevel <- c("Negative", "Untreated", "MC", "neg", "neg", "neg", "neg", "neg", "neg", "neg", "B-cell")
comp_relevel2 <- c("Positive", "Treated", "NS", "pos", "pos", "pos", "pos", "pos", "pos", "pos", "T-cell")
#### automation in progress
for (i in 1:length(comp)){
  var <- comp[i]
  dds <- paste("dds_", var, sep = "")
  res <- paste("res_", var, sep = "")
  res_Ordered <- paste("res_Ordered", var, sep = "")
  dds <- DESeqDataSetFromMatrix(countData=counts, colData=meta_nanostring, design=~comp[i])
  dds$var <- relevel(dds$var, comp_relevel[i])
  dds <- DESeq(dds)
  res <- results(dds, contrast = c(var, comp_relevel2[i], comp_relevel[i]))
  res_Ordered <- res[order(res$padj),]
  res_Ordered <- data.frame(GeneID = rownames(res_Ordered), res_Ordered)
  #write.table(res_Ordered, file = "210910_DE_UT_PRvNR.txt", quote = FALSE, sep = "\t", row.names = FALSE)
}
print("differential expression completed")
```

```{r}

######CHECK THESE FILES######

pancancer_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/NanoString_info/PanCancerIO360_R.txt"
probeannotations_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/NanoString_info/ProbeAnnotations.txt"
pathway_genes_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/Melanoma_nanostring/Results/DESeq/genes.txt"
pathway_genes_fullList_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/Melanoma_nanostring/Results/DESeq/genes_fullList.txt"
pancancer <- read.table(pancancer_path, sep = "\t", header = TRUE, row.names = 1, stringsAsFactors = F)
probeannotations <- read.table(probeannotations_path, sep = "\t", header = TRUE, row.names = 1, stringsAsFactors = F)
pathway_genes <- read.table(pathway_genes_path, sep = "\t", header = TRUE, row.names = NULL, stringsAsFactors = F)
pathway_genes_fullList <- read.table(pathway_genes_fullList_path, sep = "\t", header = TRUE, row.names = NULL, stringsAsFactors = F)

```

```{r}

  # take log2, zscore of counts
  log2_ncounter_output_file <- log2(ncounter_output_file)
  zscore_ncounter_output_file <- apply(log2_ncounter_output_file, 1, zscore)
  zscore_ncounter_output_file <- data.frame(t(zscore_ncounter_output_file))



## gene expression heatmaps
counts$HL_29<-NULL
colnames(meta_nanostring)[which(names(meta_nanostring) == "B-cell.rich/T-cell.rich/indeterminate")] <- "dom_celltype"

dds <- DESeqDataSetFromMatrix(countData=counts, colData=meta_nanostring, design=~MHCI_IHC_abbrev)
dds$MHCI_IHC_abbrev <- relevel(dds$MHCI_IHC_abbrev, "neg")
dds <- DESeq(dds)
res <- results(dds, contrast = c("MHCI_IHC_abbrev", "pos", "neg"))
res_Ordered <- res[order(res$padj),]
res_Ordered <- data.frame(GeneID = rownames(res_Ordered), res_Ordered)
# filter significant DE genes, reorder by log2FoldChange
p_val <- 0.01
DE_filter <- filter(res_Ordered, as.numeric(res_Ordered$padj) < p_val)
DE_filter <- DE_filter[order(DE_filter$log2FoldChange, decreasing=TRUE),]
# subset DE genes
all_subset <- subset(counts, rownames(counts) %in% rownames(DE_filter))
# reorder subset by log2FoldChange
all_subset <- all_subset[order(match(rownames(all_subset), rownames(DE_filter))), ]
# take log2, zscore of counts
## I MOVED THIS UP ##
all_log2_counts <- log2(all_subset)
all_z_score <- apply(all_log2_counts, 1, zscore)
all_z_score <- data.frame(t(all_z_score))
# prep column annotations
#annot_col <- data.frame(EBV_final = meta_nanostring$EBV_final, row.names = rownames(meta_nanostring))
annot_col <- data.frame(MHCI_IHC_abbrev = meta_nanostring$MHCI_IHC_abbrev, row.names = rownames(meta_nanostring))
#annot_color <- list(EBV_final = c(Negative="grey", Positive="red", unknown="grey2"))
# prep row annotations

# color, column labels
my_breaks=seq(-2,2, by=0.1)
my_color <- colorpanel(n=length(my_breaks)-1,low="blue",mid="grey88",high="red")
pdf("MHCI_IHC_abbrev_legend", 10, 10)
pheatmap(all_z_score, legend = T, color = my_color, breaks = my_breaks, show_colnames = T, show_rownames = T, 
         cluster_rows = F, cluster_cols = T, annotation_col = annot_col, annotation_row = pancancer[,-14],
         annotation_colors = annot_color, annotation_legend = T, angle_col = 90, border_color = "white", 
         main = paste("p-adjusted <", p_val, "sorted by L2FC"))
dev.off()

##########
# prep row annotations
row_annot <- data.frame(matrix(0, ncol=6, nrow=770))
row.names(row_annot) <- row.names(counts)
row_annot <- row_annot %>% rename(gamma = X1, alpha = X2, il2 = X3, antigen = X4, dysfunction = X5, TLS = X6)
row_annot$gamma <- row.names(row_annot) %in% pathway_genes$gamma_genes
row_annot$alpha <- row.names(row_annot) %in% pathway_genes$alpha_genes
row_annot$il2 <- row.names(row_annot) %in% pathway_genes$il2_genes
row_annot$antigen <- row.names(row_annot) %in% pathway_genes$antigen_presentation
row_annot$dysfunction <- row.names(row_annot) %in% pathway_genes$dysfunction
row_annot$TLS <- row.names(row_annot) %in% pathway_genes$TLS
cols <- sapply(row_annot, is.logical)
row_annot[,cols] <- lapply(row_annot[,cols], as.numeric)

annot_color <- list(EBV_final = c(Negative="cyan4", Positive="deeppink4", unknown="grey70"), 
                    Treatment = c(Untreated="cyan4", Treated="deeppink4", unknown="grey70"), 
                    Diagnosis_abbrev = c(MC="cyan4", NS="deeppink4", LR="plum4", NLPHL="wheat4", unknown="grey70"), 
                    MHCI_IHC_abbrev = c(neg="cyan4", pos="deeppink4", cytoplasmic="pink3", "n/a"="grey70", unknown="grey70"), 
                    MHCII_IHC_abbrev = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    B2M_IHC_abbrev = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    PDL1_IHC_abbrev = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    PDL2_IHC = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    CT2A_IHC_abbrev = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    LMP1_IHC_abbrev = c(neg="cyan4", pos="deeppink4", unknown="grey70"), 
                    dom_celltype = c("B-cell"="cyan4", "T-cell"="deeppink4", indeterminate="grey70"), 
                    Release_of_Cancer_Cell_Antigens = c("+"="grey10", "-"="grey90"), 
                    Cancer_Antigen_Presentation = c("+"="grey10", "-"="grey90"), 
                    T_cell_Priming_and_Activation = c("+"="grey10", "-"="grey90"), 
                    Immune_Cell_Localization_to_Tumors = c("+"="grey10", "-"="grey90"), 
                    Stromal_Factors = c("+"="grey10", "-"="grey90"), 
                    Recognition_of_Cancer_Cells_by_T_cells = c("+"="grey10", "-"="grey90"), 
                    Killing_of_Cancer_Cells = c("+"="grey10", "-"="grey90"), 
                    Myeloid_Cell_Activity = c("+"="grey10", "-"="grey90"), 
                    NK_Cell_Activity = c("+"="grey10", "-"="grey90"), 
                    Cell_Cycle_and_Proliferation = c("+"="grey10", "-"="grey90"), 
                    Tumor_Intrinsic_Factors = c("+"="grey10", "-"="grey90"), 
                    Immunometabolism = c("+"="grey10", "-"="grey90"), 
                    Common_Signaling_Pathways = c("+"="grey10", "-"="grey90"))
##########


# create heatmap
#pdf("lesion_response_heatmap.pdf", 10, 10)
pheatmap(all_z_score, legend = T, color = my_color, breaks = my_breaks, show_colnames = T, show_rownames = T, 
         cluster_rows = F, cluster_cols = T, annotation_col = annot_col, 
         annotation_colors = annot_color, annotation_legend = T, angle_col = 90, border_color = "white", 
         main = paste("p-adjusted <", p_val, "sorted by L2FC"))
#dev.off()

```

```{r}
cibersort<-read.table("/Users/pourmalm/Desktop/CIBERSORTx_Job7_Results.txt", header = TRUE, sep = "\t")
celldive<-read.table("/Users/pourmalm/Desktop/celldive.txt", header = TRUE, row.names = 1, sep = "\t")
celldive<-t(celldive)
celldive<-data.frame(celldive)
celldive$H16_5562<-NULL
celldive$HL_20<-NULL
celldive$HL_26<-NULL
celldive$HL_28<-NULL
celldive$HL_32<-NULL
# replace cell dive ID with sample ID in celldive
i1 <- match(colnames(celldive), meta_nanostring$CellDive_ID)
i2 <- !is.na(i1)
colnames(celldive)[i2] <- rownames(meta_nanostring)[i1[i2]]
celldive <- celldive %>% select(order(colnames(celldive)))
# compute percentages for celldive
for (i in 1:ncol(celldive)){
  celldive[11, i] <- sum(celldive[,i], na.rm=T)
}
for (i in 1:ncol(celldive)){
  for (j in 1:nrow(celldive)-1){
    celldive[j,i]<-celldive[j,i]/celldive[11,i]
  }
}
celldive <- celldive[-11,]
celldive<-data.frame(t(celldive))
rownames(cibersort) <- cibersort$Sample_ID
cibersort$Sample_ID <- NULL
celldive$Other <- NULL
identical(rownames(cibersort), rownames(celldive))
identical(colnames(cibersort), colnames(celldive))
colnames(cibersort) <- c("T_CD4_A", "T_CD8_A", "T_Reg_A", "T_Other_A", "B_All_A", "NK_All_A", "Macrophage_A", "Plasma_A", 
                         "Leukocyte_Other_A")
bind2 <- cbind(celldive, cibersort)
identical(rownames(bind2), rownames(meta_nanostring))
bind2$main_pop_IHC <- meta_nanostring$`B-cell.rich/T-cell.rich/indeterminate`

##

# create list of celldive, cibersort pairs
plot_list2 <- data.frame(matrix(ncol = 2, nrow = ncol(celldive)))
colnames(plot_list2) <- c('celldive', 'cibersort')
plot_list2$celldive <- colnames(celldive)
plot_list2$cibersort <- colnames(cibersort)
# create list of all the scatter plots with gene/protein correlation at the sample level and print
plts <- list()
for (i in 1:nrow(plot_list2)){
  print(i)
  p <- ggscatter(bind2, x = plot_list2$celldive[i], y = plot_list2$cibersort[i], 
            add ="reg.line", # Add regressin line, 
            add.params = list(color = "grey", fill = "lightgray"), # Customize reg. line
            conf.int = TRUE, 
            xlab = paste(plot_list2$celldive[i], "(mpIF)"), 
            ylab =  paste(str_remove(plot_list2$cibersort[i], "_A"), "(cibersort)")) # Add confidence interval 
  p <- p + stat_cor(method = "pearson", cor.coef.name	="R")
  print(p)
  plts[[i]] <- p  # add each plot into plot list
}
#multiplot(plotlist = plts, cols = 6)
ggsave("celldive-cibersort_correl.pdf", 
       gridExtra::marrangeGrob(grobs = plts, nrow = 3, ncol = 3),
       path = "/Users/pourmalm/Desktop", 
       device = "pdf", 
       width = 300, 
       height = 300, 
       units = "mm")

#add main cell population color by IHC to scatter plot (B Cell)
p <- ggscatter(bind2, x = "B_All", y = "B_All_A",
            color = "main_pop_IHC", 
            add ="reg.line", # Add regressin line, 
            add.params = list(color = "grey", fill = "lightgray"), # Customize reg. line
            conf.int = TRUE, 
            xlab = "B_All (mpIF)", 
            ylab =  "B_All (cibersort)", 
            palette = c("B-cell" = "green", "T-cell" = "blue", "indeterminate" = "grey20")) # Add confidence interval 
p <- p + stat_cor(method = "pearson", cor.coef.name	="R")
ggsave("b_cell_celldive-cibersort-ihc.pdf", p)

# pool all T cell pops
bind2$T_All <- bind2$T_CD4 + bind2$T_CD8 + bind2$T_Reg + bind2$T_Other
bind2$T_All_A <- bind2$T_CD4_A + bind2$T_CD8_A + bind2$T_Reg_A + bind2$T_Other_A

#add main cell population color by IHC to scatter plot (B Cell)
p <- ggscatter(bind2, x = "T_All", y = "T_All_A",
            color = "main_pop_IHC", 
            add ="reg.line", # Add regressin line, 
            add.params = list(color = "grey", fill = "lightgray"), # Customize reg. line
            conf.int = TRUE, 
            xlab = "T_All (mpIF)", 
            ylab =  "T_All (cibersort)", 
            palette = c("B-cell" = "green", "T-cell" = "blue", "indeterminate" = "grey20")) # Add confidence interval 
p <- p + stat_cor(method = "pearson", cor.coef.name	="R")
ggsave("t_cell_celldive-cibersort-ihc.pdf", p)

```
```


```{r}

#explore tumor functional marker expression
markersF <- c("B2M.PCT", "B7H3.PCT", "CD27.PCT", "CD40.PCT", "CD40L.PCT", "ICOS.PCT", "IDO1.PCT", "KI67.PCT", "LAG3.PCT", "MHCI.PCT", "MHCII.PCT", "MRC1.PCT", "PD1.PCT", "PDL1.PCT", "TGM2.PCT", "TIM3.PCT", "VISTA.PCT")

plot_boxplot <- function(dataframe, samples_meta, marker){
  marker <- enquo(marker)
  dataframe %>% 
    select(-c("FOV_ID", "HRS Cells", "N.Markers")) %>% 
    select(-contains(".N")) %>% 
    left_join(samples_meta %>% 
                select(Patient_ID, CellDive_ID), by = "CellDive_ID") %>% 
    select(-CellDive_ID) %>% 
    relocate(Patient_ID) %>% 
    ggplot(aes(x = Patient_ID, y = !!marker)) + 
    geom_boxplot() + 
    geom_jitter(shape = 16, size = 1, position = position_jitter(0.2)) + 
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
}

plot_grid(plot_boxplot(pTumor_FOV, samples, B2M.PCT), 
          plot_boxplot(pTumor_FOV, samples, B7H3.PCT), 
          plot_boxplot(pTumor_FOV, samples, CD27.PCT), 
          plot_boxplot(pTumor_FOV, samples, CD40.PCT))

plot_grid(plot_boxplot(pTumor_FOV, samples, CD40L.PCT), 
          plot_boxplot(pTumor_FOV, samples, ICOS.PCT), 
          plot_boxplot(pTumor_FOV, samples, IDO1.PCT), 
          plot_boxplot(pTumor_FOV, samples, KI67.PCT))

plot_grid(plot_boxplot(pTumor_FOV, samples, LAG3.PCT), 
          plot_boxplot(pTumor_FOV, samples, MHCI.PCT), 
          plot_boxplot(pTumor_FOV, samples, MHCII.PCT), 
          plot_boxplot(pTumor_FOV, samples, MRC1.PCT))

plot_grid(plot_boxplot(pTumor_FOV, samples, PD1.PCT), 
          plot_boxplot(pTumor_FOV, samples, PDL1.PCT), 
          plot_boxplot(pTumor_FOV, samples, TGM2.PCT), 
          plot_boxplot(pTumor_FOV, samples, TIM3.PCT), 
          plot_boxplot(pTumor_FOV, samples, VISTA.PCT))
          
```

```{r}

##gene:protein (cell dive - total DAPI cells) correlation

# list of genes in cell dive panel
gene_list <- c("B2M", "CD276", "CD14", "CD163", "MS4A1", "CD27", "CD3D", "CD3E","CD3G", "CD30", "CD40", "CD40LG", "CD4", 
               "CD45RA", "CD45RB", "CD45RO", "NCAM1", "CD8A", "CD8B", "FOXP3", "ICOS", "IDO1", "MKI67", "LAG3", 
               "HLA-A", "HLA-B", "HLA-C", "HLA-DRB5", "HLA-DRB5", "HLA-DRA", "MRC1", "IRF4", "PDCD1", "CD274", "HAVCR2", 
               "VSIR")
common_genes <- data.frame(t(subset(zscore_counts, rownames(zscore_counts) %in% gene_list)))
# replace cell dive ID with sample ID in counts in pDAPI and pTumor_sample and pTumor_FOV
p1 <- match(rownames(pDAPI), samples$CellDive_ID)
p2 <- !is.na(p1)
rownames(pDAPI)[p2] <- rownames(samples)[p1[p2]]
pDAPI <- pDAPI %>% arrange(row.names(pDAPI))
t1 <- match(rownames(pTumor_sample), samples$CellDive_ID)
t2 <- !is.na(t1)
rownames(pTumor_sample)[t2] <- rownames(samples)[t1[t2]]
pTumor_sample <- pTumor_sample %>% arrange(row.names(pTumor_sample))
# drop samples without nanostring data in pDAPI and pTumor
pDAPI_nanostring <- pDAPI[complete.cases(samples$Nanostring_ID), ]
#pTumor_nanostring <- pTumor[complete.cases(samples$Nanostring_ID), ]
identical(rownames(pDAPI_nanostring), rownames(common_genes))
bind <- data.frame(cbind(pDAPI_nanostring, common_genes))
# create list of protein-gene pairs
plot_list <- data.frame(matrix(ncol = 2, nrow = ncol(common_genes)))
colnames(plot_list) <- c('protein_x', 'gene_y')
plot_list$gene_y <- colnames(common_genes)
plot_list$protein_x <- c("B2M.PCT", "CD14.PCT", "CD163.PCT", "CD27.PCT", "PDL1.PCT", "B7H3.PCT", "CD3.PCT", "CD3.PCT", 
                         "CD3.PCT", "CD4.PCT", "CD40.PCT", "CD40L.PCT", "CD45.PCT", "CD45.PCT", "CD45.PCT", "CD8.PCT", 
                         "CD8.PCT", "FOXP3.PCT", "TIM3.PCT", "MHCI.PCT", "MHCI.PCT", "MHCI.PCT", "MHCII.PCT", "MHCII.PCT",  
                         "ICOS.PCT", "IDO1.PCT", "MUM1.PCT", "LAG3.PCT", "KI67.PCT", "MRC1.PCT", "CD20.PCT", "CD56.PCT", 
                         "PD1.PCT", "VISTA.PCT")
# create list of all the scatter plots with gene/protein correlation at the sample level and print
plts <- list()
for (i in 1:nrow(plot_list)){
  p <- ggscatter(bind, x = plot_list$protein_x[i], y = plot_list$gene_y[i], 
            add ="reg.line", # Add regressin line, 
            add.params = list(color = "grey", fill = "lightgray"), # Customize reg. line
            conf.int = TRUE, 
            xlab = paste(str_remove(plot_list$protein_x[i], ".PCT")), 
            ylab =  paste(plot_list$gene_y[i])) # Add confidence interval 
  p <- p + stat_cor(method = "pearson", cor.coef.name	="R")
  print(p)
  plts[[i]] <- p  # add each plot into plot list
}
#multiplot(plotlist = plts, cols = 6)
ggsave("gene_protein-celldive_correl.pdf", 
       gridExtra::marrangeGrob(grobs = plts, nrow = 6, ncol = 6),
       path = "/Users/pourmalm/Desktop", 
       device = "pdf", 
       width = 300, 
       height = 300, 
       units = "mm")
```

```{r}

##protein (IHC - tumor cells): protein (cell dive - tumor cells) correlation

# list of overlapping IHC/cell dive proteins
celldive_protein_list <- c("B2M.PCT", "MHCI.PCT", "MHCII.PCT", "PDL1.PCT")
celldive <- pTumor[, celldive_protein_list]
ihc_protein_list <- c("B2M_IHC_abbrev", "MHCI_IHC_abbrev", "MHCII_IHC_abbrev", "PDL1_IHC_abbrev")
ihc <- samples[, ihc_protein_list]
ihc <- ihc[1:44, ]
# drop 3 samples in IHC that have 0 tumor cells in celldive (this will eventually be corrected)
ihc <- subset(ihc, rownames(ihc) %in% rownames(celldive))
identical(rownames(celldive), rownames(ihc))
bind <- data.frame(cbind(celldive, ihc))
bind$Sample_ID <- rownames(bind)
# create barplots
plts2 <- list()
for (i in 1:length(celldive_protein_list)){
  bind <- bind %>% arrange(factor(bind[, i+length(celldive_protein_list)], 
                                  levels = c("pos", "cytoplasmic", "neg", "unknown")), desc(bind[, i]))
  bind$Sample_ID <- factor(bind$Sample_ID, levels = bind$Sample_ID)
  p <- ggplot(data=bind, aes(x = Sample_ID, y = bind[, i], fill = bind[, i+length(celldive_protein_list)])) + 
    geom_bar(stat="identity") + 
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
    labs(fill = paste(str_remove(ihc_protein_list[i], "_abbrev"))) + 
    ylab("% tumor cells positive (cell dive)")
  print(p)
  ggsave(paste(str_remove(ihc_protein_list[i], "_abbrev"), "_celldive_correl", ".pdf", sep=""), p, 
         path = "/Users/pourmalm/Desktop", device = "pdf")
  plts2[[i]] <- p
}
```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*.

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Cmd+Shift+K* to preview the HTML file). 

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.


_____

library(openxlsx)
library(ggplot2)

tumor_counts_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/Hodgkin_celldive/211122_tumor_functional_marker_counts_per_sample.xlsx"
tumor_counts_path <- "/Users/pourmalm/Documents/Mellinghoff_lab/Data/Hodgkin_celldive/test.xlsx"

tumor_counts <- read.xlsx(tumor_counts_path, colNames = TRUE, rowNames = FALSE)

#violin plot of tumor functional markers
ggplot(test, aes(x=marker, y=PCT, fill=CellDive_ID)) + geom_dotplot(binaxis='y', stackdir='center',
                                                          position=position_dodge(1),dotsize=0.5)

ggplot(test, y=PCT)) + geom_violin() + geom_dotplot(binaxis='y', stackdir='center', dotsize=0.3)

______



################################### other hodgkin code ###################################

#celltype_barplot.R code
library(ggplot2)
library(dplyr)

count_path <- "/Users/pourmalm/Desktop/counts.txt"
count <- read.table(count_path, sep = "\t", header = TRUE, row.names = NULL, stringsAsFactors = F)

order <- c("T_CD4", "T_CD8", "T_Reg4", "T_Reg8", "T_RegDN", "T_CD4/CD8", "T_Null", "B_All", "NK", "NKT", 
           "NKT_CD8", "NKT_CD4", "Macrophage_MHCIIpos", "Macrophage_MHCIIneg", "Macrophage_Other", 
           "Plasma", "Granulocyte", "Leukocyte_Other")
colors <- c("palegreen", "steelblue1", "palegreen4", "steelblue4", "khaki", "khaki", "khaki4", "yellow1", 
            "orchid", "orchid2", "orchid3", "orchid4", "firebrick", "firebrick3", "firebrick4", "bisque", "bisque4",
            "gray69")

# summarize overall counts
overall <- count
overall$CellType <- NULL
overall$SubType <- NULL
overall$Total <- NULL
overall <- overall %>% group_by(Overall) %>% summarise_each(funs(sum))
overall <- melt(overall)

pdf("overall_count.pdf", 20, 10)
ggplot(overall, aes(fill=factor(Overall, level=c("Tumor", "Immune", "Other")), y=value, x=variable)) + 
  geom_bar(position="stack", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1), axis.text=element_text(size=18),
        axis.title=element_text(size=20,face="bold"), legend.text=element_text(size=14)) +
  scale_fill_manual(values = c("red", "palegreen3", "grey")) + labs(fill = "Cell Type") + xlab("Patient ID") + ylab("Total Count")
dev.off()

pdf("overall_percent.pdf", 20, 10)
ggplot(overall, aes(fill=factor(Overall, level=c("Tumor", "Immune", "Other")), y=value, x=variable)) + 
  geom_bar(position="fill", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1), axis.text=element_text(size=18),
        axis.title=element_text(size=20,face="bold"), legend.text=element_text(size=14)) +
  scale_fill_manual(values = c("red", "palegreen3", "grey")) + labs(fill = "Cell Type") + xlab("Patient ID") + ylab("Percent")
dev.off()

sub <- count
sub$Overall <- NULL
sub$CellType <- NULL
sub$Total <- NULL
sub <- sub %>% group_by(SubType) %>% summarise_each(funs(sum))
sub <- sub[-c(3,13,21),]
sub <- melt(sub)

pdf("sub_count.pdf", 20, 10)
ggplot(sub, aes(fill=factor(SubType, level=order), y=value, x=variable)) + 
  geom_bar(position="stack", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1), axis.text=element_text(size=18),
        axis.title=element_text(size=20,face="bold"), legend.text=element_text(size=14)) +
  scale_fill_manual(values = colors) + labs(fill = "Cell Type") + xlab("Patient ID") + ylab("Total Count") + 
  scale_y_continuous(name="Fluorescent intensity/arbitrary units", labels = scales::comma)
dev.off()

pdf("sub_percent.pdf", 20, 10)
ggplot(sub, aes(fill=factor(SubType, level=order), y=value, x=variable)) + 
  geom_bar(position="fill", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1), axis.text=element_text(size=18),
        axis.title=element_text(size=20,face="bold"), legend.text=element_text(size=14)) +
  scale_fill_manual(values = colors) + labs(fill = "Cell Type") + xlab("Patient ID") + ylab("Percent")
dev.off()



